### PR TITLE
[AWS] Add identity info when check fails

### PR DESCRIPTION
**Pull request by Michaelvll**
This adds more observability for #2451 

Tested (run the relevant ones):

- [ ] Code formatting: `bash format.sh`
- [ ] Any manual or new tests for this PR (please specify below)
- [ ] All smoke tests: `pytest tests/test_smoke.py` 
- [ ] Relevant individual smoke tests: `pytest tests/test_smoke.py::test_fill_in_the_name` 
- [ ] Backward compatibility tests: `bash tests/backward_comaptibility_tests.sh`

### CODE REVIEW

File: sky/clouds/service_catalog/data_fetchers/fetch_aws.py [hunk 1]
```diff
@@ -136,8 +136,8 @@ def _get_availability_zones(region: str) -> Optional[pd.DataFrame]:
                     'Failed to retrieve availability zone. '
                     'Please ensure that the `ec2:DescribeAvailabilityZones` '
                     'action is enabled for your AWS account in IAM. '
-                    'Ref: https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_DescribeAvailabilityZones.html'  # pylint: disable=line-too-long
-                ) from None
+                    'Ref: https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_DescribeAvailabilityZones.html.\n'  # pylint: disable=line-too-long
+                    f'Details: {e}') from None
```
Comment (concretevitamin) [inline sky/clouds/service_catalog/data_fetchers/fetch_aws.py:140]: Confused when we want to format_exception() and when we don't.

File: sky/clouds/aws.py [hunk 2]
```diff
@@ -486,9 +486,10 @@ def check_credentials(cls) -> Tuple[bool, Optional[str]]:
             aws_catalog.get_default_instance_type()
         except RuntimeError as e:
             return False, (
-                'Failed to fetch the availability zones for the account. It is '
-                'likely due to permission issues, please check the minimal '
-                'permission required for AWS: https://skypilot.readthedocs.io/en/latest/cloud-setup/cloud-permissions/aws.html'  # pylint: disable=
+                'Failed to fetch the availability zones for the account '
+                f'{identity}. It is likely due to permission issues, please '
```
Comment (concretevitamin) [inline sky/clouds/aws.py:490]:
```suggestion
                f'{identity!r}. It is likely due to permission issues, please '
```

File: sky/clouds/service_catalog/data_fetchers/fetch_aws.py [hunk 3]
```diff
@@ -136,8 +136,8 @@ def _get_availability_zones(region: str) -> Optional[pd.DataFrame]:
                     'Failed to retrieve availability zone. '
```
Comment (concretevitamin) [inline sky/clouds/service_catalog/data_fetchers/fetch_aws.py:136]:
```suggestion
                    'Failed to retrieve availability zones. '
```

**Review by concretevitamin — APPROVED** [review]

File: sky/clouds/aws.py (hunk 2 above)
Comment (Michaelvll) [reply sky/clouds/aws.py:490]: This should be a list of strings. We probably don't want to add `!r`

File: sky/clouds/service_catalog/data_fetchers/fetch_aws.py (hunk 1 above)
Comment (Michaelvll) [reply sky/clouds/service_catalog/data_fetchers/fetch_aws.py]: Ahh, added the format

File: sky/clouds/service_catalog/data_fetchers/fetch_aws.py [hunk 4]
```diff
@@ -133,10 +134,12 @@ def _get_availability_zones(region: str) -> Optional[pd.DataFrame]:
         elif e.response['Error']['Code'] == 'UnauthorizedOperation':
             with ux_utils.print_exception_no_traceback():
                 raise RuntimeError(
-                    'Failed to retrieve availability zone. '
+                    'Failed to retrieve availability zones. '
                     'Please ensure that the `ec2:DescribeAvailabilityZones` '
                     'action is enabled for your AWS account in IAM. '
-                    'Ref: https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_DescribeAvailabilityZones.html'  # pylint: disable=line-too-long
+                    'Ref: https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_DescribeAvailabilityZones.html.\n'  # pylint: disable=line-too-long
```
Comment (TRT-BradleyB) [inline sky/clouds/service_catalog/data_fetchers/fetch_aws.py:140]: Some more detail on this raise would be good - currently just recieve the error: "Details: [botocore.exceptions.ClientError] An error occurred (UnauthorizedOperation) when calling the DescribeAvailabilityZones operation: You are not authorized to perform this operation."

despite: `aws ec2 describe-availability-zones` working fine.

Can you log the account at this point also?

Comment (Michaelvll) [reply sky/clouds/service_catalog/data_fetchers/fetch_aws.py:140]: Thanks for the comment @TRT-BradleyB! The upper-level caller in the `aws.py` should log the identity information above. Do you see that message when running `sky check`?

Comment (Michaelvll) [reply sky/clouds/service_catalog/data_fetchers/fetch_aws.py:140]: Hmm, this is weird. I did see the following messages when running `sky check`:
```console
$ sky check                                                     
Checking credentials to enable clouds for SkyPilot.
  Checking AWS...I 08-24 10:09:34 aws_catalog.py:79] Fetching availability zones mapping for AWS...
  AWS: disabled          
    Reason: Failed to fetch the availability zones for the account ['my-account', 'my-account']. It is likely due to permission issues, please check the minimal permission required for AWS: https://skypilot.readthedocs.io/en/latest/cloud-setup/cloud-permissions/aws.html
    Details: [builtins.RuntimeError] Failed to retrieve availability zones. Please ensure that the `ec2:DescribeAvailabilityZones` action is enabled for your AWS account in IAM. Ref: https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_DescribeAvailabilityZones.html. Details: [botocore.exceptions.ClientError] ...
```

Just to confirm, is the error you shown from `sky check`?

Comment (Michaelvll) [reply sky/clouds/service_catalog/data_fetchers/fetch_aws.py]: Thanks for confirming! Merging this for now, and we can continue the discussion in #2451 for the availability zones permission issue.

**Pull request merged**